include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -112,6 +122,16 @@ class random_access_iterator
 
     //!\brief Use default deconstructor.
     ~random_access_iterator() = default;
+
+    //!\brief Copy constructor for const version from non-const version.
+    template <typename t>
+    //!\cond
+        requires std::is_same_v<std::remove_const_t<container_type>, t> &&
+                 std::is_same_v<container_type, std::add_const_t<t>>
+    //!\endcond
+    constexpr random_access_iterator(random_access_iterator<t> const & rhs) noexcept :
```

Comment:
I don't think we need assignment. Actually, if you have copy elision (or even just proper move-assignment) of your type and a custom constructor from a different type, I don't think you ever need an additional assignment operator nowadays, do you? I mean, it will